### ISSUE TITLE
still two issues after update

### ISSUE THREAD

**Issue by cfl2005** [logpai/loglizer#31]
Hi，
    Thank you very much for the update. I tried it the first time after the update.Successfully analyzed the log abnormal:
Train validation:
====== Evaluation summary ======
Precision: 0.966, recall: 0.365, F1-measure: 0.530

Test validation:
====== Evaluation summary ======
Precision: 0.967, recall: 0.561, F1-measure: 0.710

But I still have two questions：
1、How can I get anomaly_label.csv in demo file?
2、How do I get specific exception log information after the exception detection is completed?

best wishes ，thank you！

**Comment by jimzhu** [comment]
Thanks for the comment. The demo log and anomaly_label.csv is in the repo, under 'data/' folder. The second question is a quite practical requirement. We cannot provide this at current stage, but we will plan it in future. You can add the meta info during data loading for inspecting anomaly log messages. That is, adding the blk_id and raw log at `load_HDFS()`.

**Comment by cfl2005** [comment]
Thanks ! I mean, how do I generate anomaly_label.csv. For example, if it is zookeeper, how should I generate the corresponding anomaly_label.csv?

**Comment by jimzhu** [comment]
OK. The label file is manually labelled. For zookeeper, we have no such groundtruth labels yet. We marked those datasets with `(labeled)` in the loghub page.

**Comment by cfl2005** [comment]
Thanks!Due to work reasons, I'm doing log anomaly detection, so I want to know the labeling rules.

**Comment by jimzhu** [comment]
It is defined by some type of rules defined by experts. It is oringinated from the the paper "Mining Console Logs for Large-Scale System Problem Detection".

**Comment by Rufaida94** [comment]
I am a bit confused about the anomaly_label.csv file, can I still do invariants mining without the anomaly_label.csv file? if I am not interested in measuring the correctness at this point (precision, F1-measure, recall)..

**Comment by jimzhu** [comment]
Yes, of course. It will be follow the steps:
1. x_train = load_data() # load event sequences
2. x_train = fit_transform(x_train) # get event count vectors
3. model.fit(x_train) # train the model to get invariants
4. model.predict(x_train) # anomaly prediction results (but not evaluation on the accuracy)

I will update a demo file later for this case.

**Comment by Rufaida94** [comment]
Great! thanks a lot. I will give a try now and hope to see the demo soon.

**Comment by jimzhu** [comment]
@Rufaida94  The demo file PCA_demo_without_label.py has been updated.

**Comment by Rufaida94** [comment]
OK, thanks. let me know when you update InvariantsMiner_demo.py

**Comment by jimzhu** [comment]
You can follow the PCA demo to change the code  to InvariantsMiner straightforwardly.

**Comment by Rufaida94** [comment]
OK, I will do that.

**Comment by Rufaida94** [comment]
also, it would be great if you can add some text to explain the meaning of the mined invariants

(Mined 11 invariants: {(14,): [1], (0, 2): [1.0, -1.0], (0, 12): [-15.0, 1.0], (1, 6): [1.0, -3.0], (1, 8): [-1.0, 1.0], (1, 10): [-1.0, 1.0], (1, 13): [1.0, -1.0], (3, 4): [1.0, -1.0], (3, 7): [1.0, -1.0], (3, 9): [-101.0, 1.0], (3, 11): [-2.0, 1.0]})


it is hard to interpret them now.

**Comment by jimzhu** [comment]
Yes, the docs are going to be updated. The invariant mined the event ids and the correponding weights. For example  (0, 2): [1.0, -1.0], denotes E0 * 1.0 + E2 * (-1.0) = 0.
